File: js/pricd_prijm.js
!function(e) {
    var o = function(e) {
        return 12 == e.length ? dqdm = e.substring(1, 5) : dqdm = e.substring(0, 4),
        "2102" != dqdm && "3302" != dqdm && "3502" != dqdm && "3702" != dqdm && "4403" != dqdm && (dqdm = dqdm.substring(0, 2) + "00"),
        dqdm
    }
      , r = function(e) {
        return e.length >= 12 ? e.substring(0, 10) : e.substring(0, 2)
    }
      , p = function(e) {
        return e.length + 5 * e.length
    };
    e.extend({
        prijm: function(r, p, a, t, s, c, m) {
            var n = o(r);
            switch (n) {
            case "1100":
                m = e.encrypt(r + e.moveTo(m)).toUpperCase(),
                "0" == swjgmcft && (m = e.encrypt(r + e.moveTo(m) + r).toUpperCase());
                break;
            case "1200":
                m = e.encrypt(r + e.moveTo(m) + p).toUpperCase();
                break;
            case "1300":
                m = e.encrypt(a + e.moveTo(m) + e.moveTo(r)).toUpperCase();
                break;
            case "1400":
                m = e.encrypt(e.moveTo(m) + s).toUpperCase(),
                "0" == swjgmcft && (m = e.encrypt(e.moveTo(m) + s + m).toUpperCase());
                break;
            case "1500":
                m = e.encrypt(e.moveTo(m) + e.moveTo(r) + p).toUpperCase();
                break;
            case "2100":
                m = e.encrypt(r + p + m).toUpperCase();
                break;
            case "2102":
                m = e.encrypt(r + e.moveTo(p) + m).toUpperCase();
                break;
            case "2200":
                m = e.encrypt(r + m + e.moveTo(m)).toUpperCase();
                break;
            case "2300":
                m = e.encrypt(s + m).toUpperCase();
                break;
            case "3100":
                m = e.encrypt(e.moveTo(m)).toUpperCase(),
                "0" == swjgmcft && (m = e.encrypt(e.moveTo(m) + p).toUpperCase());
                break;
            case "3200":
                m = e.encrypt(r + m).toUpperCase();
                break;
            case "3300":
                m = e.encrypt(p + m).toUpperCase();
                break;
            case "3302":
                m = e.encrypt(e.moveTo(p) + m).toUpperCase();
                break;
            case "3400":
                m = e.encrypt(r + e.moveTo(a) + m).toUpperCase();
                break;
            case "3500":
                m = e.encrypt(n + e.moveTo(a) + m).toUpperCase();
                break;
            case "3502":
                m = e.encrypt(n + e.moveTo(m) + r).toUpperCase();
                break;
            case "3600":
                m = e.encrypt(e.encrypt(m) + p + a).toUpperCase();
                break;
            case "3700":
                m = e.encrypt(m + e.moveTo(n) + r).toUpperCase(),
                "0" == swjgmcft && (m = e.encrypt(e.moveTo(m) + n).toUpperCase());
                break;
            case "3702":
                m = e.encrypt(e.encrypt(p) + e.moveTo(m) + r).toUpperCase();
                break;
            case "4100":
                m = e.encrypt(e.moveTo(m) + e.moveTo(r) + m).toUpperCase();
                break;
            case "4200":
                m = e.encrypt(a + e.moveTo(a) + m).toUpperCase();
                break;
            case "4300":
                m = e.encrypt(e.moveTo(a) + m + p).toUpperCase();
                break;
            case "4400":
                m = e.encrypt(e.moveTo(r) + e.moveTo(s) + m).toUpperCase();
                break;
            case "4403":
                m = e.encrypt(n + e.moveTo(m) + m).toUpperCase();
                break;
            case "4500":
                m = e.encrypt(n + e.moveTo(r) + m + a).toUpperCase();
                break;
            case "4600":
                m = e.encrypt(p + e.moveTo(r) + m + a).toUpperCase();
                break;
            case "5000":
                m = e.encrypt(r + e.moveTo(r) + r + m).toUpperCase();
                break;
            case "5100":
                m = e.encrypt(a + e.moveTo(r) + p + m).toUpperCase();
                break;
            case "5200":
                m = e.encrypt(t + e.moveTo(r) + m).toUpperCase();
                break;
            case "5300":
                m = e.encrypt(a + e.moveTo(r) + m).toUpperCase(),
                "0" == swjgmcft && (m = e.encrypt(e.moveTo(m) + r).toUpperCase());
                break;
            case "5400":
                m = e.encrypt(n + e.moveTo(r) + m).toUpperCase();
                break;
            case "6100":
                m = e.encrypt(n + m + e.moveTo(p) + e.moveTo(r)).toUpperCase();
                break;
            case "6200":
                m = e.encrypt(a + e.moveTo(p) + m).toUpperCase();
                break;
            case "6300":
                m = e.encrypt(r + e.moveTo(n) + m).toUpperCase();
                break;
            case "6400":
                m = e.encrypt(p + e.moveTo(s) + m).toUpperCase();
                break;
            case "6500":
                m = e.encrypt(t + e.moveTo(m) + m).toUpperCase()
            }
            return m
        },
        pricd: function(a, t, s) {
            var c = o(a);
            switch (c) {
            case "1100":
                s = e.encrypt(a + e.moveTo(t) + s).toUpperCase(),
                "0" == swjgmcft && (s = e.encrypt(e.moveTo(s) + a).toUpperCase());
                break;
            case "1200":
                s = e.encrypt(a + e.moveTo(a + t) + s).toUpperCase();
                break;
            case "1300":
                s = e.encrypt(a + e.moveTo(a.substring(2, 3) + t) + s).toUpperCase(),
                "0" == swjgmcft && (s = e.encrypt(e.moveTo(s) + a + t).toUpperCase());
                break;
            case "1400":
                s = e.encrypt(a + e.moveTo(t) + s).toUpperCase();
                break;
            case "1500":
                s = e.encrypt(t + e.moveTo(t + s) + s.length).toUpperCase();
                break;
            case "2100":
                s = e.encrypt(s + e.moveTo(c) + s.toLowerCase()).toUpperCase();
                break;
            case "2102":
                s = e.encrypt(e.moveTo(a + t) + e.moveTo(a) + s).toUpperCase();
                break;
            case "2200":
                s = e.encrypt(a.length + t.length + e.moveTo(s) + s).toUpperCase();
                break;
            case "2300":
                s = e.encrypt(e.encrypt(a) + e.moveTo(c) + e.moveTo(s.substring(10, 15))).toUpperCase(),
                "0" == swjgmcft && (s = e.encrypt(e.moveTo(s) + t + c).toUpperCase());
                break;
            case "3100":
                s = e.encrypt(t + e.moveTo(a) + s + a).toUpperCase();
                break;
            case "3200":
                s = e.encrypt(e.moveTo(c) + e.moveTo(c) + s).toUpperCase();
                break;
            case "3300":
                s = e.encrypt(r(a) + e.moveTo(c) + s).toUpperCase();
                break;
            case "3302":
                s = e.encrypt(e.moveTo(a) + e.moveTo(r(c)) + s).toUpperCase();
                break;
            case "3400":
                s = e.encrypt(a + r(e.moveTo(c)) + r(s)).toUpperCase(),
                "0" == swjgmcft && (s = e.encrypt(e.moveTo(s) + t + r(s)).toUpperCase());
                break;
            case "3500":
                s = e.encrypt(r(a) + e.moveTo(c) + s).toUpperCase();
                break;
            case "3502":
                s = e.encrypt(r(a + e.moveTo(c)) + e.moveTo(s)).toUpperCase();
                break;
            case "3600":
                s = e.encrypt(p(a) + e.moveTo(s) + p(s)).toUpperCase();
                break;
            case "3700":
                s = e.encrypt(p(a) + e.moveTo(c) + r(s + a)).toUpperCase();
                break;
            case "3702":
                s = e.encrypt(e.moveTo(a + e.moveTo(c)) + s).toUpperCase();
                break;
            case "4100":
                s = e.encrypt(a + e.moveTo(e.moveTo(c) + s) + p(t)).toUpperCase();
                break;
            case "4200":
                s = e.encrypt(t + e.moveTo(e.moveTo(r(s)) + s) + r(t)).toUpperCase();
                break;
            case "4300":
                s = e.encrypt(e.moveTo(a) + e.moveTo(c) + s + p(s)).toUpperCase();
                break;
            case "4400":
                s = e.encrypt(e.encrypt(a + e.moveTo(c)) + e.moveTo(s)).toUpperCase();
                break;
            case "4403":
                s = e.encrypt(e.moveTo(a + e.moveTo(s)) + e.encrypt(s)).toUpperCase();
                break;
            case "4500":
                s = e.encrypt(e.moveTo(e.moveTo(a) + e.moveTo(c) + s)).toUpperCase(),
                "0" == swjgmcft && (s = e.encrypt(e.moveTo(s) + t + e.moveTo(c)).toUpperCase());
                break;
            case "4600":
                s = e.encrypt(e.moveTo(r(a) + p(t)) + e.moveTo(a) + s).toUpperCase();
                break;
            case "5000":
                s = e.encrypt(e.moveTo(a) + p(a) + e.moveTo(t) + s).toUpperCase();
                break;
            case "5100":
                s = e.encrypt(p(a) + e.moveTo(s + a) + r(s)).toUpperCase();
                break;
            case "5200":
                s = e.encrypt(e.moveTo(s) + e.moveTo(s + t) + s).toUpperCase();
                break;
            case "5300":
                s = e.encrypt(e.moveTo(a + e.moveTo(t)) + s).toUpperCase();
                break;
            case "5400":
                s = e.encrypt(p(e.moveTo(a)) + e.moveTo(c) + s).toUpperCase();
                break;
            case "6100":
                s = e.encrypt(r(a + e.moveTo(c)) + e.moveTo(s) + r(a + t)).toUpperCase();
                break;
            case "6200":
                s = e.encrypt(e.moveTo(a + e.moveTo(a + t + s)) + s).toUpperCase();
                break;
            case "6300":
                s = e.encrypt(e.moveTo(e.moveTo(r(a) + c)) + a + p(a + t)).toUpperCase(),
                "0" == swjgmcft && (s = e.encrypt(e.moveTo(s) + t + a + p(a + t)).toUpperCase());
                break;
            case "6400":
                s = e.encrypt(e.moveTo(a) + s + e.moveTo(c + p(a)) + s).toUpperCase();
                break;
            case "6500":
                s = e.encrypt(e.moveTo(t) + e.moveTo(e.moveTo(a + s)) + s).toUpperCase()
            }
            return s
        }
    })
}(jQuery);

File: scripts/pricd_prijm.py
import execjs
from scripts.encrypt import encrypt
from scripts.gen_moveTo import moveTo

js = r'''
    var o = function (e) {
            return 12 == e.length ? dqdm = e.substring(1, 5) : dqdm = e.substring(0, 4), "2102" != dqdm && "3302" != dqdm && "3502" != dqdm && "3702" != dqdm && "4403" != dqdm && (dqdm = dqdm.substring(0, 2) + "00"), dqdm
        };
'''
ctx = execjs.compile(js)


def p(e):
    return len(e)+5*len(e)


def r(e):
    if len(e) >= 12:
        return e[0:10]
    else:
        return e[0:2]


def pricd(a, t, s):
    c = ctx.call("o", a)
    if c == "1100":
        s = encrypt(a + moveTo(t) + s).upper()
        s = encrypt(moveTo(s) + a).upper()

    elif c == "1200":
        s = encrypt(a + moveTo(a + t) + s).upper()

    elif c == "1300":
        s = encrypt(a + moveTo(a[2:3] + t) + s).upper()
        s = encrypt(moveTo(s) + a + t).upper()

    elif c == "1400":
        s = encrypt(a + moveTo(t) + s).upper()

    elif c == "1500":
        s = encrypt(t + moveTo(t + s) + len(s)).upper()

    elif c == "2100":
        s = encrypt(s + moveTo(c) + s.lower()).upper()

    elif c == "2102":
        s = encrypt(moveTo(a + t) + moveTo(a) + s).upper()

    elif c == "2200":
        s = encrypt(len(a) + len(t) + moveTo(s) + s).upper()

    elif c == "2300":
        s = encrypt(encrypt(a) + moveTo(c) + moveTo(s[10, 15])).upper()
        s = encrypt(moveTo(s) + t + c).upper()

    elif c == "3100":
        s = encrypt(t + moveTo(a) + s + a).upper()

    elif c == "3200":
        s = encrypt(moveTo(c) + moveTo(c) + s).upper()

    elif c == "3300":
        s = encrypt(r(a) + moveTo(c) + s).upper()

    elif c == "3302":
        s = encrypt(moveTo(a) + moveTo(r(c)) + s).upper()

    elif c == "3400":
        s = encrypt(a + r(moveTo(c)) + r(s)).upper()
        s = encrypt(moveTo(s) + t + r(s)).upper()

    elif c == "3500":
        s = encrypt(r(a) + moveTo(c) + s).upper()

    elif c == "3502":
        s = encrypt(r(a + moveTo(c)) + moveTo(s)).upper()

    elif c == "3600":
        s = encrypt(p(a) + moveTo(s) + p(s)).upper()

    elif c == "3700":
        s = encrypt(p(a) + moveTo(c) + r(s + a)).upper()

    elif c == "3702":
        s = encrypt(moveTo(a + moveTo(c)) + s).upper()

    elif c == "4100":
        s = encrypt(a + moveTo(moveTo(c) + s) + p(t)).upper()

    elif c == "4200":
        s = encrypt(
            t + moveTo(moveTo(r(s)) + s) + r(t)).upper()

    elif c == "4300":
        s = encrypt(moveTo(a) + moveTo(c) + s + p(s)).upper()

    elif c == "4400":
        s = encrypt(encrypt(a + moveTo(c)) + moveTo(s)).upper()

    elif c == "4403":
        s = encrypt(moveTo(a + moveTo(s)) + encrypt(s)).upper()

    elif c == "4500":
        s = encrypt(moveTo(moveTo(a) + moveTo(c) + s)).upper()
        s = encrypt(moveTo(s) + t + moveTo(c)).upper()

    elif c == "4600":
        s = encrypt(moveTo(r(a) + p(t)) + moveTo(a) + s).upper()

    elif c == "5000":
        s = encrypt(moveTo(a) + p(a) + moveTo(t) + s).upper()

    elif c == "5100":
        s = encrypt(p(a) + moveTo(s + a) + r(s)).upper()

    elif c == "5200":
        s = encrypt(moveTo(s) + moveTo(s + t) + s).upper()

    elif c == "5300":
        s = encrypt(moveTo(a + moveTo(t)) + s).upper()

    elif c == "5400":
        s = encrypt(p(moveTo(a)) + moveTo(c) + s).upper()

    elif c == "6100":
        s = encrypt(r(a + moveTo(c)) + moveTo(s) + r(a + t)).upper()

    elif c == "6200":
        s = encrypt(moveTo(a + moveTo(a + t + s)) + s).upper()

    elif c == "6300":
        s = encrypt(moveTo(moveTo(r(a) + c)) + a + p(a + t)).upper()
        s = encrypt(moveTo(s) + t + a + p(a + t)).upper()

    elif c == "6400":
        s = encrypt(moveTo(a) + s + moveTo(c + p(a)) + s).upper()

    elif c == "6500":
        s = encrypt(moveTo(t) + moveTo(moveTo(a + s)) + s).upper()
    return s


def prijm(r, p, a, t, s, c, m):
    n = ctx.call("o", r)
    if n == "1100":
        m = encrypt(r + moveTo(m)).upper()
        m = encrypt(r + moveTo(m) + r).upper()

    elif n == "1200":
        m = encrypt(r + moveTo(m) + p).upper()

    elif n == "1300":
        m = encrypt(a + moveTo(m) + moveTo(r)).upper()

    elif n == "1400":
        m = encrypt(moveTo(m) + s).upper()
        m = encrypt(moveTo(m) + s + m).upper()

    elif n == "1500":
        m = encrypt(moveTo(m) + moveTo(r) + p).upper()

    elif n == "2100":
        m = encrypt(r + p + m).upper()

    elif n == "2102":
        m = encrypt(r + moveTo(p) + m).upper()

    elif n == "2200":
        m = encrypt(r + m + moveTo(m)).upper()

    elif n == "2300":
        m = encrypt(s + m).upper()

    elif n == "3100":
        m = encrypt(moveTo(m)).upper()
        m = encrypt(moveTo(m) + p).upper()

    elif n == "3200":
        m = encrypt(r + m).upper()

    elif n == "3300":
        m = encrypt(p + m).upper()

    elif n == "3302":
        m = encrypt(moveTo(p) + m).upper()

    elif n == "3400":
        m = encrypt(r + moveTo(a) + m).upper()

    elif n == "3500":
        m = encrypt(n + moveTo(a) + m).upper()

    elif n == "3502":
        m = encrypt(n + moveTo(m) + r).upper()

    elif n == "3600":
        m = encrypt(encrypt(m) + p + a).upper()

    elif n == "3700":
        m = encrypt(m + moveTo(n) + r).upper()
        m = encrypt(moveTo(m) + n).upper()

    elif n == "3702":
        m = encrypt(encrypt(p) + moveTo(m) + r).upper()

    elif n == "4100":
        m = encrypt(moveTo(m) + moveTo(r) + m).upper()

    elif n == "4200":
        m = encrypt(a + moveTo(a) + m).upper()

    elif n == "4300":
        m = encrypt(moveTo(a) + m + p).upper()

    elif n == "4400":
        m = encrypt(moveTo(r) + moveTo(s) + m).upper()

    elif n == "4403":
        m = encrypt(n + moveTo(m) + m).upper()

    elif n == "4500":
        m = encrypt(n + moveTo(r) + m + a).upper()

    elif n == "4600":
        m = encrypt(p + moveTo(r) + m + a).upper()

    elif n == "5000":
        m = encrypt(r + moveTo(r) + r + m).upper()

    elif n == "5100":
        m = encrypt(a + moveTo(r) + p + m).upper()

    elif n == "5200":
        m = encrypt(t + moveTo(r) + m).upper()

    elif n == "5300":
        m = encrypt(a + moveTo(r) + m).upper()
        m = encrypt(moveTo(m) + r).upper()

    elif n == "5400":
        m = encrypt(n + moveTo(r) + m).upper()

    elif n == "6100":
        m = encrypt(n + m + moveTo(p) + moveTo(r)).upper()

    elif n == "6200":
        m = encrypt(a + moveTo(p) + m).upper()

    elif n == "6300":
        m = encrypt(r + moveTo(n) + m).upper()

    elif n == "6400":
        m = encrypt(p + moveTo(s) + m).upper()

    elif n == "6500":
        m = encrypt(t + moveTo(m) + m).upper()
    return m


if __name__ == "__main__":
    print(prijm("011001900311", "26540678", "20190708", "316342",
                "2019-08-30 14:29:48", "ek", "05EEA095BA29535C6CEDC47059A771F7"))
